lib/puppet/resource_api/transport.rb

```diff
@@ -0,0 +1,20 @@
+require 'puppet/resource_api/transport_schema_def'
+# Remote target transport API
+module Puppet::ResourceApi::Transport
+  class << self
+    attr_reader :transports
```

Comment:
exposing the transports like this will allow folks to directly interact with the hash and add/modify/delete entries without validation. there is no need for the accessor yet.